core/src/test/scala/akka/persistence/r2dbc/query/EventsByPersistenceIdSpec.scala

```diff
@@ -0,0 +1,222 @@
+        case Current =>
+          query.currentEventsByPersistenceId(pid, from, to)
+      }
+
+    def assertFinished(probe: TestSubscriber.Probe[_], liveShouldFinish: Boolean = false): Unit =
+      queryType match {
+        case Live if !liveShouldFinish =>
+          probe.expectNoMessage()
+          probe.cancel()
+        case _ =>
+          probe.expectComplete()
+      }
+
+    s"$queryType eventsByPersistenceId" should {
+      "populates timestamp offset" in {
+        val pid = nextPid()
+        val persister = testKit.spawn(Persister(pid))
+        val probe = testKit.createTestProbe[Done]()
+        persister ! Persister.PersistWithAck("e-1", probe.ref)
+        probe.expectMessage(Done)
+
+        val sub = doQuery(pid, 0, Long.MaxValue)
+          .runWith(TestSink())
+          .request(1)
+
+        sub.expectNextPF {
+          case EventEnvelope(TimestampOffset(_, _, seen), `pid`, 1, "e-1") if seen == Map(pid -> 1) =>
+        }
+
+        assertFinished(sub)
+      }
+
+      "return all events then complete" in {
+        val pid = nextPid()
+        val persister = testKit.spawn(Persister(pid))
+        val probe = testKit.createTestProbe[Done]()
+        val events = (1 to 20).map { i =>
+          val payload = s"e-$i"
+          persister ! PersistWithAck(payload, probe.ref)
+          probe.expectMessage(Done)
+          payload
+        }
+
+        val sub = doQuery(pid, 0, Long.MaxValue)
+          .map(_.event)
+          .runWith(TestSink())
+
+        sub
+          .request(events.size + 1)
+          .expectNextN(events.size)
+
+        assertFinished(sub)
+      }
+
+      "only return sequence nrs requested" in {
+        val pid = nextPid()
+        val persister = testKit.spawn(Persister(pid))
+        val probe = testKit.createTestProbe[Done]()
+        val events = (1 to 20).map { i =>
+          val payload = s"e-$i"
+          persister ! PersistWithAck(payload, probe.ref)
+          probe.expectMessage(Done)
+          payload
+        }
+
+        val sub = doQuery(pid, 0, 5)
+          .map(_.event)
+          .runWith(TestSink())
+
+        sub
+          .request(events.size + 1)
+          .expectNextN(events.take(5))
+
+        assertFinished(sub, liveShouldFinish = true)
+      }
+
+      "allow querying for a single event" in {
+        val pid = nextPid()
+        val persister = testKit.spawn(Persister(pid))
+        val probe = testKit.createTestProbe[Done]()
+
+        (1 to 3).map { i =>
+          val payload = s"e-$i"
+          persister ! PersistWithAck(payload, probe.ref)
+          probe.expectMessage(Done)
+          payload
+        }
+
+        val sub = doQuery(pid, 2, 2)
+          .map(_.event)
+          .runWith(TestSink())
+
+        val event = sub
+          .request(2)
+          .expectNext()
+        event should ===("e-2")
+
+        assertFinished(sub, liveShouldFinish = true)
+      }
+
+      "include metadata" in {
```

Comment:
took most of this test from the Spanner plugin, but this test of metadata is new